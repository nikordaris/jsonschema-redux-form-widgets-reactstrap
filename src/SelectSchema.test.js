import React from 'react';
import renderer from 'react-test-renderer';
import inputFields from './index';
import { Provider } from 'react-redux';
import { reduxForm, reducer as formReducer } from 'redux-form';
import { createStore, combineReducers } from 'redux';

const { SingleSelectInput } = inputFields;

const testInputFieldSnapshot = options => () => {
  const rootReducers = combineReducers({ form: formReducer });
  const store = createStore(rootReducers);
  const WrappedComponent = reduxForm({ form: 'MyForm' })(SingleSelectInput);
  const tree = renderer
    .create(
      <Provider store={store}>
        <WrappedComponent {...options} renderSchema={() => <div />} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
};

describe('Render SelectSchema', () => {
  it(
    'Should render SingleSelectInput',
    testInputFieldSnapshot({
      schema: {
        id: 'SingleSelectInput',
        title: 'SingleSelectInput',
        type: 'number',
        oneOf: [
          { title: 'foo', const: 1, description: 'foo description' },
          { title: 'bar', const: 2, description: 'bar description' }
        ]
      },
      name: 'SingleSelectInput'
    })
  );

  it(
    'Should render SingleSelectInput with only values',
    testInputFieldSnapshot({
      schema: {
        id: 'SingleSelectInput',
        title: 'SingleSelectInput',
        type: 'string',
        oneOf: [{ const: 'foo' }, { const: 'bar' }]
      },
      name: 'SingleSelectInput'
    })
  );

  it(
    'Should render SingleSelectInput with groups',
    testInputFieldSnapshot({
      schema: {
        id: 'SingleSelectInput',
        title: 'SingleSelectInput',
        type: 'number',
        oneOf: [
          {
            title: 'Foo Group',
            oneOf: [
              { title: 'foo', const: 1, description: 'foo description' },
              { title: 'foo2', const: 2, description: 'foo2 description' }
            ]
          },
          {
            title: 'Bar Group',
            oneOf: [
              { title: 'bar', const: 1, description: 'bar description' },
              { title: 'bar2', const: 2, description: 'bar2 description' }
            ]
          }
        ]
      },
      name: 'SingleSelectInput'
    })
  );
});
