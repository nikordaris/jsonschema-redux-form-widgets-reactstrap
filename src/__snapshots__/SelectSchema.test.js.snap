// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SelectSchema Should render SingleSelectInput 1`] = `
<div
  className="card-0-1 card"
>
  <div
    className="header-0-2 card-header"
  >
    SingleSelectInput
  </div>
  <div
    className="card-block"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value=""
    >
      <option
        disabled={true}
        value=""
      >
        Select 
        SingleSelectInput
      </option>
      <option
        value="bar"
      >
        bar
      </option>
      <option
        value="foo"
      >
        foo
      </option>
    </select>
  </div>
</div>
`;

exports[`Render SelectSchema Should render SingleSelectInput with groups 1`] = `
<div
  className="card-0-9 card"
>
  <div
    className="header-0-10 card-header"
  >
    SingleSelectInput
  </div>
  <div
    className="card-block"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value=""
    >
      <option
        disabled={true}
        value=""
      >
        Select 
        SingleSelectInput
      </option>
      <option
        value="Bar Group"
      >
        Bar Group
      </option>
      <option
        value="Foo Group"
      >
        Foo Group
      </option>
    </select>
  </div>
</div>
`;

exports[`Render SelectSchema Should render SingleSelectInput with only values 1`] = `
<div
  className="card-0-5 card"
>
  <div
    className="header-0-6 card-header"
  >
    SingleSelectInput
  </div>
  <div
    className="card-block"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value=""
    >
      <option
        disabled={true}
        value=""
      >
        Select 
        SingleSelectInput
      </option>
      <option
        value="bar"
      >
        bar
      </option>
      <option
        value="foo"
      >
        foo
      </option>
    </select>
  </div>
</div>
`;
